packages/compare-slider-block/src/CompareSliderBlock.tsx

```diff
@@ -0,0 +1,397 @@
+        setSlotWithUploadInProgress(slot);
+
+        setDroppedFiles(files);
+    };
+
+    const handleAssetDelete = (key: string, id: number) => {
+        deleteAssetIdsFromKey(key, [id]);
+    };
+
+    const openAssetChooser = (slot: SliderImageSlot) => {
+        appBridge.openAssetChooser(
+            async (result) => {
+                if (slot === SliderImageSlot.First) {
+                    setIsFirstAssetLoading(true);
+                } else {
+                    setIsSecondAssetLoading(true);
+                }
+                updateAssetIdsFromKey(slot === SliderImageSlot.First ? 'firstAsset' : 'secondAsset', [result[0].id]);
+                appBridge.closeAssetChooser();
+            },
+            {
+                objectTypes: [AssetChooserObjectType.ImageVideo],
+                extensions: FileExtensionSets['Images'],
+            }
+        );
+    };
+
+    const calculateAutoImageHeight = (): number | undefined => {
+        if (!firstAsset || !secondAsset) {
+            return 0;
+        }
+
+        const currentSliderWidth = sliderRef.current?.clientWidth;
+
+        if (!currentSliderWidth) {
+            return undefined;
+        }
+
+        const assetWithSmallerAspectRatio =
+            firstAsset[0].width / firstAsset[0].height > secondAsset[0].width / firstAsset[0].height
+                ? secondAsset[0]
+                : firstAsset[0];
+
+        return Math.round(
+            (assetWithSmallerAspectRatio.height * currentSliderWidth) / assetWithSmallerAspectRatio.width
+        );
+    };
+
+    const getImageHeight = (): number | undefined => {
+        if (hasCustomHeight) {
+            return parseInt(customHeight);
+        }
+
+        if (height !== Height.Auto) {
+            return heightMap[height];
+        }
+
+        return calculateAutoImageHeight();
+    };
+
+    const getBorderStyle = () => {
+        return hasBorder ? { borderWidth, borderColor: toRgbaString(borderColor), borderStyle } : {};
+    };
+
+    const getBorderRadius = () => {
+        if (hasRadius) {
+            return radiusValue;
+        }
+
+        return radiusStyleMap[radiusChoice];
+    };
+
+    const renderSliderItem = (slot: SliderImageSlot) => {
+        return (
+            <div>
+                <div style={{ background: hasBackground ? toRgbaString(backgroundColor) : undefined }}>
+                    <img
+                        src={slot === SliderImageSlot.First ? firstAssetPreviewUrl : secondAssetPreviewUrl}
+                        alt={slot === SliderImageSlot.First ? firstAssetTitle : secondAssetTitle}
+                        className="tw-w-full tw-object-cover"
+                        style={{ height: getImageHeight() }}
+                    />
+                </div>
+
+                {renderStrikethrough(slot)}
+                {!isEditing && renderLabel(slot)}
+            </div>
+        );
+    };
+
+    const renderStrikethrough = (slot: SliderImageSlot) => {
```

Comment:
consider to put everything into the StrikethroughWrapper directly, so the CompareSliderBlock.tsx would be much cleaner